src/python/ensembl/io/genomio/genome_metadata/compare_genome_stats.py

```diff
@@ -0,0 +1,214 @@
+    """
+    # Prepare counts to be comparable
+    core_biotypes = core.get("genes", {}).get("biotypes", {})
+
+    # We want a value for:
+    # protein_coding = same
+    # pseudogene = all pseudogene biotypes
+    # total = same
+    # other = number of misc_RNA
+
+    ncbi_counts = {
+        "protein_coding": ncbi.get("protein_coding", 0),
+        "pseudogene": ncbi.get("pseudogene", 0),
+        "total_genes": ncbi.get("total", 0),
+        "other": ncbi.get("other", 0),
+    }
+
+    # Add all pseudogenes
+    num_pseudogenes = 0
+    for name, num in core_biotypes.items():
+        if re.match(".*pseudogen.*", name):
+            num_pseudogenes += num
+
+    # Others? misc_mRNA (or anything similar?)
+    num_others = core_biotypes.get("misc_RNA", 0)
+
+    core_counts = {
+        "protein_coding": core_biotypes.get("protein_coding", 0),
+        "pseudogene": num_pseudogenes,
+        "total_genes": core.get("genes", {}).get("total", 0),
+        "other": num_others,
+    }
+
+    return _diff_dicts(ncbi_counts, core_counts)
+
+
+def compare_stats(ncbi: Dict, core: Dict) -> Dict[str, Any]:
+    """Compare stats from NCBI and our core.
+
+    Args:
+        ncbi: Dict of stats from NCBI.
+        core: Dict of stats from the core.
+
+    Returns:
+        Dict: Each count from NCBI, from Core, and their diff.
+    """
+
+    ncbi_annotation_stats = ncbi.get("annotation_info", {}).get("stats", {}).get("gene_counts", {})
+    core_assembly_stats = core.get("assembly_stats", {})
+    core_annotation_stats = core.get("annotation_stats", {})
+
+    comp: Dict[str, Any] = {
+        "assembly_diff": compare_assembly(ncbi, core_assembly_stats),
+    }
+    if core_annotation_stats and ncbi_annotation_stats:
+        comp["annotation_diff"] = compare_annotation(ncbi_annotation_stats, core_annotation_stats)
+
+    return comp
+
+
+class InputSchema(argschema.ArgSchema):
+    """Input arguments expected by this script."""
+
+    # Server parameters
+    ncbi_stats = argschema.fields.InputFile(required=True, metadata={"description": "NCBI json file"})
+    core_stats = argschema.fields.InputFile(required=True, metadata={"description": "Core json file"})
+
+
+def main() -> None:
+    """Main script entry-point."""
+    mod = argschema.ArgSchemaParser(schema_type=InputSchema)
+    args = mod.args
+
+    with open(args["ncbi_stats"]) as ncbi_fh:
+        try:
+            ncbi_stats = json.load(ncbi_fh)["reports"][0]
+        except KeyError:
+            ncbi_stats = {}
+    with open(args["core_stats"]) as core_fh:
+        core_stats = json.load(core_fh)
+    all_stats = compare_stats(ncbi_stats, core_stats)
+
+    if args.get("output_json"):
+        output_file = Path(args.get("output_json"))
+        with output_file.open("w") as output_fh:
+            output_fh.write(json.dumps(all_stats, indent=2, sort_keys=True))
+    else:
+        print(json.dumps(all_stats, indent=2, sort_keys=True))
+
+
+if __name__ == "__main__":
```

Comment:
not required ?

Reply:
I'd like to keep that if only for testing